src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts

```diff
@@ -67,8 +77,19 @@ export class MarkdownTextTransformer {
 				endLineNumber: selection.startLineNumber
 			};
 
-			let beginInsertedText = getStartTextToInsert(type);
-			let endInsertedText = getEndTextToInsert(type);
+			let beginInsertedText: string;
+			let endInsertedText: string;
+			if (type === MarkdownButtonType.IMAGE) {
+				if (!this._selectImageDialog) {
+					this._selectImageDialog = this._instantiationService.createInstance(SelectImageDialog);
+					this._selectImageDialog.render();
+				}
+				let selectImageOptions = await this._selectImageDialog.open();
+				beginInsertedText = selectImageOptions.imageHtml;
```

Comment:
What happens if they cancel the dialog?

Reply:
@Charles-Gagnon --> @corivera wired this up. Upon testing, if you close the dialog, no tag is added to the text or markdown, and the cursor remains where it was prior to clicking the Image button.